sql-tests/testdata/drop_metric.sql

```diff
@@ -0,0 +1,127 @@
+    END
+  FROM _prom_catalog.ids_epoch 
+  LIMIT 1;
+
+ CALL _prom_catalog.finalize_metric_creation();
+END$$;
+-- Ingestion complete
+
+-- Checking state of the ingested data prior to drop attempts
+SELECT ok(count(*) = 4) FROM prom_data.test;
+SELECT ok(count(*) = 3, 'none of the series should be removed yet') FROM _prom_catalog.series;
+SELECT ok(count(*) = 0, 'none of the series should be marked for deletion') FROM _prom_catalog.series WHERE delete_epoch IS NOT NULL;
+SELECT ok(count(*) = 3, 'none of the labels should deleted yet') FROM _prom_catalog.label where key='name1';
+
+-- Dropping the data
+CREATE FUNCTION asserts_before_deletion(msg TEXT)
+ RETURNS SETOF TEXT
+ LANGUAGE plpgsql VOLATILE AS
+$fnc$
+BEGIN
+ RETURN NEXT is(count(*), 2::BIGINT, msg) FROM prom_data.test;
+ RETURN NEXT is(count(*), 3::BIGINT, msg || ': none of the series should be removed yet') FROM _prom_catalog.series;
+ RETURN NEXT is(count(*), 1::BIGINT, msg || ': one series should be marked for deletion') FROM _prom_catalog.series WHERE delete_epoch IS NOT NULL;
+ RETURN NEXT is(count(*), 3::BIGINT, msg || ': none of the labels should deleted yet') FROM _prom_catalog.label where key='name1';
+ RETURN;
+END;
+$fnc$;
+
+-- The first attempt to drop the chunks.
+CALL _prom_catalog.drop_metric_chunks('prom_data', 'test', E'2009-11-11 00:00:05+00');
+SELECT asserts_before_deletion('after the first timestamp');
+-- Attempting to drop chunks while incrementally moving `run_at` by an hour
+-- reruns shouldn't change anything until the dead series are actually dropped
+CALL _prom_catalog.drop_metric_chunks('prom_data', 'test', E'2009-11-11 00:00:05+00');
+SELECT asserts_before_deletion('after iter 0');
+CALL _prom_catalog.drop_metric_chunks('prom_data', 'test', E'2009-11-11 00:00:05+00', now() + '1 hours');
+SELECT asserts_before_deletion('after iter 1');
+CALL _prom_catalog.drop_metric_chunks('prom_data', 'test', E'2009-11-11 00:00:05+00', now() + '2 hours');
+SELECT asserts_before_deletion('after iter 2');
+CALL _prom_catalog.drop_metric_chunks('prom_data', 'test', E'2009-11-11 00:00:05+00', now() + '3 hours');
+SELECT asserts_before_deletion('after iter 3');
+CALL _prom_catalog.drop_metric_chunks('prom_data', 'test', E'2009-11-11 00:00:05+00', now() + '4 hours');
+SELECT asserts_before_deletion('after iter 4');
+
+
+CREATE FUNCTION asserts_after_deletion(msg TEXT)
+ RETURNS SETOF TEXT
+ LANGUAGE plpgsql VOLATILE AS
+$fnc$
+BEGIN
+ RETURN NEXT is(count(*), 2::BIGINT, msg) FROM prom_data.test;
+ RETURN NEXT is(count(*), 2::BIGINT, msg || ': one series should be removed') FROM _prom_catalog.series;
+ RETURN NEXT is(count(*), 0::BIGINT, msg || ': no series should be marked for deletion') FROM _prom_catalog.series WHERE delete_epoch IS NOT NULL;
+ RETURN NEXT is(count(*), 2::BIGINT, msg || ': unused labels should deleted') FROM _prom_catalog.label where key='name1';
+ RETURN;
+END;
+$fnc$;
+
+-- Now is the time when it should actually drop the chunks
```

Comment:
Could you explicitly comment why we expect the chunks to be dropped, and series to be removed?

I _think_ that it has to do with the fact that the current_epoch > delete_epoch + 4, and because we already ran 4 deletions above, we expect the fifth one to drop.

Reply:
done